tests/pipelines/test_pipelines_common.py

```diff
@@ -795,6 +800,135 @@ def test_cfg(self):
         assert out_cfg.shape == out_no_cfg.shape
 
 
+@is_staging_test
+class PipelinePushToHubTester(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls._token = TOKEN
+        HfFolder.save_token(TOKEN)
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            delete_repo(token=cls._token, repo_id="test-pipeline")
```

Comment:
Usually what we do in `huggingface_hub` is to define a repo_id that is unique per test. Something like `repo_id=f"test-diffusers-pipeline-{uuid.uuid4()}"`. You can create this repo_id in `def setUp(self)` and delete in the test (or `def tearDown(self)`).

The goal is to have perfectly independent tests: 
- if you run tests concurrently in the CI (now or in the future) you don't want tests to interfere with each other
- if you have multiple CIs running at the same time (which is very likely to happen given the number of PRs), you don't want a CI to fail because another CI deleted a repo while you were testing it.

This precaution apply to all tests below.

Reply:
One way I would do this would be to:
- remove `def setUpClass` and `def tearDownClass`
- generate the repo_id in each test case (either a org one or a "normal" one)
- delete the repo in each test case instead of trying all possibilities at the end
- if a test fail, the repo is not deleted => we don't care _that much_ since it's staging. Repos are cleaned once in a while.